benchmark/benchmain/main.go

```diff
@@ -0,0 +1,388 @@
+				caller(pos)
+				elapse := time.Since(start)
+				atomic.AddInt32(&count, 1)
+				mu.Lock()
+				s.Add(elapse)
+				mu.Unlock()
+			}
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+	stopTimer(count)
+}
+
+// Initiate main function to get settings of features.
+func init() {
+	var runUnary, runStream bool
+	var traceMode, noTraceMode bool
+	var readLatency, readTimeout string
+	var readKbps, readMtu, readMaxConcurrentCalls, readReqSizeBytes, readReqspSizeBytes intSliceType
+	var compressorMode bool
+	flag.BoolVar(&runUnary, "runUnary", false, "runUnary")
+	flag.BoolVar(&runStream, "runStream", false, "runStream")
+	flag.BoolVar(&traceMode, "traceMode", false, "traceMode")
+	flag.BoolVar(&noTraceMode, "noTraceMode", false, "noTraceMode")
+	flag.StringVar(&readLatency, "latency", "", "latency")
+	flag.StringVar(&readTimeout, "timeout", "", "timeout")
+	flag.Var(&readKbps, "kbps", "kbps")
+	flag.Var(&readMtu, "mtu", "mtu")
+	flag.Var(&readMaxConcurrentCalls, "maxConcurrentCalls", "maxConcurrentCalls")
+	flag.Var(&readReqSizeBytes, "reqSizeBytes", "reqSizeBytes")
+	flag.Var(&readReqspSizeBytes, "reqspSizeBytes", "reqspSizeBytes")
+	flag.StringVar(&memProfile, "memProfile", "", "memProfile")
+	flag.IntVar(&memProfileRate, "memProfileRate", 0, "memProfileRate")
+	flag.StringVar(&cpuProfile, "cpuProfile", "", "cpuProfile")
+	flag.BoolVar(&compressorMode, "compressorMode", false, "compressorMode")
+	flag.Parse()
+	// If no flags related to mode are set, it runs both by default.
+	if runUnary || runStream {
+		runMode[0] = runUnary
+		runMode[1] = runStream
+	}
+	// If node flags related to trace are set, it runs both mode by default.
+	if traceMode && !noTraceMode {
+		enableTrace = []bool{true}
+	}
+	if !traceMode && noTraceMode {
+		enableTrace = []bool{false}
+	}
+	if compressorMode {
+		enableCompressor = []bool{true}
+	} else {
+		enableCompressor = []bool{false}
+	}
+	// Time input formats as (time + unit).
+	readTimeFromIntSlice(&ltc, readLatency)
+	readTimeFromIntSlice(&timeout, readTimeout)
+	readIntFromIntSlice(&kbps, readKbps)
+	readIntFromIntSlice(&mtu, readMtu)
+	readIntFromIntSlice(&maxConcurrentCalls, readMaxConcurrentCalls)
+	readIntFromIntSlice(&reqSizeBytes, readReqSizeBytes)
+	readIntFromIntSlice(&respSizeBytes, readReqspSizeBytes)
+}
+
+type intSliceType []int
+
+func (intSlice *intSliceType) String() string {
+	return fmt.Sprintf("%v", *intSlice)
+}
+
+func (intSlice *intSliceType) Set(value string) error {
+	if len(*intSlice) > 0 {
+		return errors.New("interval flag already set")
+	}
+	for _, num := range strings.Split(value, ",") {
+		next, err := strconv.Atoi(num)
+		if err != nil {
+			return err
+		}
+		*intSlice = append(*intSlice, next)
+	}
+	return nil
+}
+
+func readIntFromIntSlice(values *[]int, replace intSliceType) {
+	// If not set replace in the flag, just return to run the default settings.
+	if len(replace) == 0 {
+		return
+	}
+	*values = replace
+}
+
+func readTimeFromIntSlice(values *[]time.Duration, replace string) {
```

Comment:
Should we change the name of this function since it isn't really reading from an intSlice?